### PR TITLE
Updated user guide and storage grid custom gateway

### PR DESCRIPTION
**Pull request by VaibhavPage**
- [x] Asciinema recording
- [x] Step-by-Step gateway and sensor installation guide
- [x] Refactor escalation K8 event creation 
- [x] Standard 5 field cron expression in calendar gateway 
- [x] Custom gateway for StorageGrid notifications

#102 #105 #107

### CODE REVIEW

File: controllers/sensor/controller.go [hunk 1]
```diff
@@ -173,3 +175,25 @@ func (c *SensorController) runWorker() {
 	for c.processNextItem() {
 	}
 }
+
+func escalateError(name, namespace, reason, instanceID string, kubeClientset kubernetes.Interface) error {
```
Comment (magaldima) [inline controllers/sensor/controller.go]: could we make this function public in a common package? I saw another function very similar to this

File: controllers/sensor/notification-handler.go [hunk 2]
```diff
@@ -260,7 +259,7 @@ func (se *sensorExecutionCtx) validateSignal(gatewaySignal *ss_v1alpha1.Event) (
 func (se *sensorExecutionCtx) handleSignals(w http.ResponseWriter, r *http.Request) {
 	// parse the request body which contains the cloudevents specification compliant event/signal dispatched from gateway.
 	body, err := ioutil.ReadAll(r.Body)
-	gatewaySignal := ss_v1alpha1.Event{}
+	var gatewaySignal *ss_v1alpha1.Event
```
Comment (magaldima) [inline controllers/sensor/notification-handler.go]:
```suggestion
	gatewaySignal := new(ss_v1alpha1.Event)
```

File: gateways/custom/storage-grid/main.go [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+// storageGridConfigExecutor implements ConfigExecutor interface
+type storageGridConfigExecutor struct{}
+
+// storageGridEventConfig contains configuration for storage grid sns
+type storageGridEventConfig struct {
+	Port string
+	Endpoint string
+	// Todo: add event and prefix filtering.
+	Events   []string
+	Filter   StoageGridFilter
```
Comment (magaldima) [inline gateways/custom/storage-grid/main.go]: typo in the `StorageGridFilter`

**Comment by magaldima** [comment]
Also looks like the unit tests are failing.. @VaibhavPage can you check those?

**Comment by VaibhavPage** [comment]
It weird tests fail on  namespace_test.go:27 complaining namespace mismatch of `argo-events` and `default` when there is no default namespace to begin with. Locally all tests pass.

**Comment by VaibhavPage** [comment]
Also its not running sensor related tests.

File: controllers/sensor/controller.go (hunk 1 above)
Comment (VaibhavPage) [reply controllers/sensor/controller.go]: this method was unnecessary and causing confusion, getting rid of it.

**Pull request merged**